tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -303,10 +303,57 @@ static void test_uri_parser__validate(void)
     }
 }
 
+static void test_uri_parser__unterminated_string(void)
+{
+    uri_parser_result_t ures;
+    char uri[64];
+    /* initialize with a non-null character */
+    memset(uri, 'Z', sizeof(uri));
+    char *uri_str = "coap://RIOT:test@[2001:db8::1]:5683/.well-known/core?v=1";
+
+    memcpy(uri, uri_str, strlen(uri_str));
+
+    int res = uri_parser_process(&ures, uri, strlen(uri_str));
+    TEST_ASSERT_EQUAL_INT(0, res);
+    if (!res) {
```

Comment:
If `!res` the function returns in the `TEST_ASSERT_EQUAL_INT` above, so this check is redundant.